### PR TITLE
Fixes all bugs.

### PR DESCRIPTION
**Pull request by michounn**
Currently fully working. Tested on PocketMine-MP 3.21.0

### CODE REVIEW

File: MapImageEngine/src/FaigerSYS/MapImageEngine/storage/MapImageChunk.php [hunk 1]
```diff
@@ -130,10 +130,10 @@ public function getABGR(int $x, int $y) : int {
...
 	public function setABGR(int $x, int $y, int $color) {
 		$pos = $this->getStartOffset($x, $y);
-		$this->data->buffer{$pos++} = chr($color >> 24 & 0xff);
-		$this->data->buffer{$pos++} = chr($color >> 16 & 0xff);
-		$this->data->buffer{$pos++} = chr($color >> 8  & 0xff);
-		$this->data->buffer{$pos}   = chr($color       & 0xff);
+		$this->data->buffer[$pos++] = chr($color >> 24 & 0xff);
+		$this->data->buffer[$pos++] = chr($color >> 16 & 0xff);
+		$this->data->buffer[$pos++] = chr($color >> 8  & 0xff);
+		$this->data->buffer[$pos] = chr($color       & 0xff);
```
Comment (TheNewHEROBRINEX) [inline MapImageEngine/src/FaigerSYS/MapImageEngine/storage/MapImageChunk.php:136]: Here you removed two spaces before the `=` sign that were present to make the last `chr` call aligned with the previous ones.

**Review by TheNewHEROBRINEX — CHANGES_REQUESTED** [review]

File: MapImageEngine/src/FaigerSYS/MapImageEngine/storage/MapImageChunk.php (hunk 1 above)
Comment (michoudll) [reply MapImageEngine/src/FaigerSYS/MapImageEngine/storage/MapImageChunk.php:136]: i just used code styles from pmmp. does this affect the final work of the plugin?

Comment (TheNewHEROBRINEX) [reply MapImageEngine/src/FaigerSYS/MapImageEngine/storage/MapImageChunk.php:136]: No but you shouldn't change the formatting of the code for no reason at all

Comment (michoudll) [reply MapImageEngine/src/FaigerSYS/MapImageEngine/storage/MapImageChunk.php]: i changed the formatting back, the author hasn't been here for a long time, so I think nothing will change.

File: MapImageEngine/src/FaigerSYS/MapImageEngine/storage/MapImageChunk.php [hunk 2]
```diff
@@ -105,10 +105,10 @@ public function getRGBA(int $x, int $y) : int {
...
 	public function setRGBA(int $x, int $y, int $color) {
 		$pos = $this->getStartOffset($x, $y);
-		$this->data->buffer{$pos++} = chr($color       & 0xff);
-		$this->data->buffer{$pos++} = chr($color >> 8  & 0xff);
-		$this->data->buffer{$pos++} = chr($color >> 16 & 0xff);
-		$this->data->buffer{$pos}   = chr($color >> 24 & 0xff);
+		$this->data->buffer[$pos++] = chr($color       & 0xff);
+		$this->data->buffer[$pos++] = chr($color >> 8  & 0xff);
+		$this->data->buffer[$pos++] = chr($color >> 16 & 0xff);
+		$this->data->buffer[$pos] = chr($color >> 24 & 0xff);
```
Comment (TheNewHEROBRINEX) [inline MapImageEngine/src/FaigerSYS/MapImageEngine/storage/MapImageChunk.php:111]: Also here (I didn't notice before)

**Review by TheNewHEROBRINEX — CHANGES_REQUESTED** [review]

**Review by TheNewHEROBRINEX — APPROVED** [review]

**Pull request merged**